lib/Sema/CSApply.cpp

```diff
@@ -5090,19 +5111,43 @@ namespace {
         ConstraintLocator *locator,
         SmallVectorImpl<KeyPathExpr::Component> &components) {
       auto resolvedTy = simplifyType(overload.adjustedOpenedType);
-      if (auto *property = overload.choice.getDeclOrNull()) {
-        // Key paths can only refer to properties currently.
-        auto varDecl = cast<VarDecl>(property);
-        // Key paths don't work with mutating-get properties.
-        assert(!varDecl->isGetterMutating());
-        // Key paths don't currently support static members.
-        // There is a fix which diagnoses such situation already.
-        assert(!varDecl->isStatic());
-
-        // Compute the concrete reference to the member.
-        auto ref = resolveConcreteDeclRef(property, locator);
-        components.push_back(
-            KeyPathExpr::Component::forProperty(ref, resolvedTy, componentLoc));
+      if (auto *decl = overload.choice.getDeclOrNull()) {
+        if (auto var = dyn_cast<VarDecl>(decl)) {
+          // Key paths can only refer to properties currently.
+          auto varDecl = cast<VarDecl>(decl);
+          // Key paths don't work with mutating-get properties.
+          assert(!varDecl->isGetterMutating());
+          // Key paths don't currently support static members.
```

Comment:
`Key paths don't currently support static members.` 
Can't we lift this limitation too?

```swift
enum Enum {
    case case1
    static let case2 = case1
}
let kp1 = \Enum.case1
let kp2 = \Enum.case2
```

KeyPath for a case is very similar to a static member, like enum cases can stand as witness to static protocol requirements?

Reply:
I haven't looked into static members at all yet, but my initial reaction is that they are a little different from enum cases from an implementation point of view.